### PR TITLE
README.md updated for V3.0.0

### PR DESCRIPTION
**Pull request by gitwallit**
Updated README.md by adding an introduction to FreeRTOS-Plus-TCP and moving the upgrading to V3.0.0 section to below the introduction and getting started sections (as recommended).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,7 @@
...
-
-To run the script, you should switch to the FreeRTOS-Plus-TCP directory that was created using the [Cloning this repository](#cloning-this-repository) step above.
-And then run  `python <Path/to/the/script>/GenerateOriginalFiles.py`.
+## FreeRTOS-Plus-TCP Library
+FreeRTOS-Plus-TCP is a lightweight TCP/IP stack for FreeRTOS. It provides a familiar Berkeley sockets interface, making it as simple to use and learn as possible. FreeRTOS-Plus-TCP's features and RAM footprint are fully scalable, making FreeRTOS-Plus-TCP equally applicable to smaller lower throughput microcontrollers as well as larger higher throughput microprocessors.
 
+This library has undergone static code analysis and checks for compliance with the [MISRA coding standard](https://www.misra.org.uk/).  Any deviations from the MISRA C:2012 guidelines are documented under [MISRA Deviations](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md). The library is validated for memory safety and data structure invariance through the [CBMC automated reasoning tool](https://www.cprover.org/cbmc/) for the functions that parse data originating from the network.
```
Comment (AniruddhaKanhere) [inline README.md:4]: NIT:
```suggestion
This library has undergone static code analysis and checks for compliance with the [MISRA coding standard](https://www.misra.org.uk/). Any deviations from the MISRA C:2012 guidelines are documented under [MISRA Deviations](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md). The library is validated for memory safety and data structure invariance through the [CBMC automated reasoning tool](https://www.cprover.org/cbmc/) for the functions that parse data originating from the network.
```

File: README.md [hunk 2]
```diff
@@ -26,6 +15,22 @@ If you have any questions or need assistance troubleshooting your FreeRTOS proje
 
 Also see the [Submitting a bugs/feature request](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/.github/CONTRIBUTING.md#submitting-a-bugsfeature-request) section of CONTRIBUTING.md for more details.
 
+## Updating to V3.0.0 and above
```
Comment (AniruddhaKanhere) [inline README.md:18]: Maybe we should use upgrading?

```suggestion
## Upgrading to V3.0.0 and above
```

File: README.md [hunk 3]
```diff
@@ -26,6 +15,22 @@ If you have any questions or need assistance troubleshooting your FreeRTOS proje
...
+
+### Generating pre V3.0.0 folder structure for backward compatibility:
+If you wish to continue using a version earlier than V3.0.0 i.e. continue with use of the older source code organization, a script is provided to generate the folder structure similar to [this](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/tree/f118c8415b4373e3d6e6dbd2d5a116f7eaf27b63).
```
Comment (AniruddhaKanhere) [inline README.md:22]:
```suggestion
If you wish to continue using the older source code organization (pre V3.0.0), a script is provided to generate the folder structure similar to [this](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/tree/f118c8415b4373e3d6e6dbd2d5a116f7eaf27b63).
```

**Review by AniruddhaKanhere — COMMENTED** [review]
Thanks @gitwallit for this PR.

I have a few comments/suggestions. Can you please take a look?

**Review by AniruddhaKanhere — DISMISSED** [review]

**Review by paulbartell — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -26,6 +15,22 @@ If you have any questions or need assistance troubleshooting your FreeRTOS proje
 
 Also see the [Submitting a bugs/feature request](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/.github/CONTRIBUTING.md#submitting-a-bugsfeature-request) section of CONTRIBUTING.md for more details.
 
+## Updating to V3.0.0 and above
+In versions >= 3.0.0, the folder structure of FreeRTOS-Plus-TCP has changed and the files have been broken down into smaller logically separated modules. This change makes the code more modular and conducive to unit-tests. Additionally, the source files have been moved to a `source` directory. This change requires modification of any existing project(s) to include the modified source files and directories. There are examples on how to use the new files and directory structure. For a windows simulator based example, refer to this [demo](https://github.com/FreeRTOS/FreeRTOS/tree/TCPRefactorDemo/FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator). For an example based on the Xilinx Zynq-7000, use the code in this [branch](https://github.com/aws/amazon-freertos/tree/TCPRefactorDemo) and follow these [instructions](https://docs.aws.amazon.com/freertos/latest/userguide/getting_started_xilinx.html) to build and run the demo.
```
Comment (tanmoysen) [inline README.md:19]: I'd suggest we add a section introducing v3.0.0.
"FreeRTOS-Plus-TCP V3.0.0 improves the robustness, security, and modularity of the library. FreeRTOS-Plus-TCP V3.0.0 adds comprehensive unit test coverage for all lines and branches of code, and has undergone penetration testing and protocol testing by AWS Security to reduce the exposure to security vulnerabilities. For context, protocol testing involves compliance and impairment checks for IPV4, TCP, UDP, DHCP, ARP and ICMP, which helps improve the robustness of FreeRTOS-Plus-TCP TCP/IP stack. The source code has also been restructured to make it more modular and extensible, and easier to add unit tests."

Comment (tanmoysen) [inline README.md:19]: In versions 3.0.0 or higher, the folder structure.....

File: README.md (hunk 3 above)
Comment (tanmoysen) [inline README.md]: "If you wish to continue using a version earlier than V3.0.0 i.e. continue to use your existing source code organization....

File: README.md (hunk 4 above)
Comment (gitwallit) [reply README.md]: updated

File: README.md (hunk 3 above)
Comment (gitwallit) [reply README.md]: updated

File: README.md [hunk 5]
```diff
@@ -1,31 +1,36 @@
...
 
+## Upgrading to V3.0.0 and above
+In version 3.0.0 or higher, the folder structure of FreeRTOS-Plus-TCP has changed and the files have been broken down into smaller logically separated modules. This change makes the code more modular and conducive to unit-tests. Learn more about the V3.0.0 release [here](https://freertos.org/2022/07/the-freertos-plus-tcp-library-is-now-more-robust-and-secure.html). Additionally, the source files have been moved to a `source` directory. This change requires modification of any existing project(s) to include the modified source files and directories. There are examples on how to use the new files and directory structure. For a windows simulator based example, refer to this [demo](https://github.com/FreeRTOS/FreeRTOS/tree/TCPRefactorDemo/FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator). For an example based on the Xilinx Zynq-7000, use the code in this [branch](https://github.com/aws/amazon-freertos/tree/TCPRefactorDemo) and follow these [instructions](https://docs.aws.amazon.com/freertos/latest/userguide/getting_started_xilinx.html) to build and run the demo.
```
Comment (tanmoysen) [inline README.md:19]: Pointing to the blog will worsen the user journey (back'n'forth from the blog README.) Instead, we can add a few lines to this section. Example:
"FreeRTOS-Plus-TCP V3.0.0 improves the robustness, security, and modularity of the library. FreeRTOS-Plus-TCP V3.0.0 adds comprehensive unit test coverage for all lines and branches of code, and has undergone penetration testing and protocol testing by AWS Security to reduce the exposure to security vulnerabilities. For context, protocol testing involves compliance and impairment checks for IPV4, TCP, UDP, DHCP, ARP and ICMP, which helps improve the robustness of FreeRTOS-Plus-TCP TCP/IP stack."

**Review by tanmoysen — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**